pkg/csi/cinder/nodeserver.go

```diff
@@ -499,10 +500,19 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 		return nil, status.Error(codes.Internal, "Unable to find Device path for volume")
 	}
 
+	if ns.Cloud.GetBlockStorageOpts().RescanOnResize {
+		// comparing current volume size with the expected one
+		newSize := req.GetCapacityRange().GetRequiredBytes()
+		if err := blockdevice.RescanBlockDeviceGeometry(devicePath, volumePath, newSize); err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not verify %q volume size: %v", volumeID, err)
```

Comment:
This might not be correct error message here? the error could be of anything with related to rescan as well?

Reply:
`RescanBlockDeviceGeometry` function checks the block device size, and, if needed, triggers a `/rescan` call. If the size is wrong, then volume size verification fails.